lib/logstash/outputs/elasticsearch/http_client/manticore_adapter.rb

```diff
@@ -83,30 +83,30 @@ def perform_request(url, method, path, params={}, body=nil)
       resp
     end
 
-    def format_url(url, path=nil)
+    def format_url(url, path_and_query=nil)
       request_uri = url.clone
-
-      if path
-        # Combine the paths using the minimal # of /s
-        # First, we make sure the path is relative so URI.join does
-        # the right thing
-        relative_path = path && path.start_with?("/") ? path[1..-1] : path
-
-        if !request_uri.path
-          request_uri.path = path
-        else
-          request_uri.path = "#{request_uri.path}/#{relative_path}"
-        end
-      else
-        request_uri = request_uri.clone
-      end
-
+      
       # We excise auth info from the URL in case manticore itself tries to stick
       # sensitive data in a thrown exception or log data
       request_uri.user = nil
       request_uri.password = nil
 
-      request_uri
+      return request_uri.to_s unless path_and_query && path_and_query != nil
```

Comment:
Hmmm. There are some situations where people pass in a "". I suppose in that case the rest of the code wouldn't hurt though. I'll do that.